bolts.lic

```diff
@@ -0,0 +1,160 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#bolts
+=end
+
+custom_require.call(%w[common common-items common-crafting])
+
+class Bolts
+  include DRC
+  include DRCC
+  include DRCI
+
+  def initialize
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @bag_items = @settings.crafting_items_in_container
+    @belt = @settings.engineering_belt
+    @hometown = @settings.hometown
+    @hometown_data = get_data('crafting')['shaping'][@hometown]
+
+    arg_definitions = [
+      [
+        { name: 'type',
+          options: %w[cougar-claw boar-tusk sabertooth angiswaerd hele'la basilisk elsralael soot-stained ice-adder jagged-horn drake-fang],
+          variable: true,
+          descriptions: "Which arrow type to create?" },
+
+        { name: 'lumber',
+          regex: /\w+/i,
+          variable: true,
+          description: 'Type of lumber used.' },
+
+        { name: 'tools',
+          regex: /yes/,
+          optional: true,
+          description: 'Get tools from clerk?'}
+      ],
+      [
+        { name: 'continue', regex: /continue/i, variable: true }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    if args.continue
+      wait_for_script_to_complete('tinker', ['continue', 'bolts'])
+      exit_tasks
+      exit
+    end
+
+    @type = args.type
+    @lumber = args.lumber
+    @tools = args.tools
+
+    get_tools if @tools
+    order_lumber
+    check_flights
+    check_glue
+    make_shafts
+    make_bolts
+    wait_for_script_to_complete('tinker', ['stow', '7', "#{@type} bolts", 'shafts', 'bolts'])
+    exit_tasks
+  end
+
+  def exit_tasks
+    clean_up
+    store_tools if @tools
+  end
+
+  def get_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'get'])
+  end
+
+  def store_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'store'])
+  end
+
+  def order_lumber
+    return if exists?("#{@lumber} lumber") || exists?('shafts')
+    wood = find_piece("#{@lumber} lumber")
+    2.times do
+      order_item(@hometown_data['stock-room'], wood)
+      bput("get #{@lumber} lumber from #{@bag}", "You get", "What were")
+      bput("combine my lumber", "You combine", "You must")
+      bput("stow lumber", "You put")
+    end
+  end
+
+  def check_flights
+    return if exists?('flights')
```

Comment:
Same note about prefixing functions -- this might be in either DRC or DRCI

Reply:
DRCI added to these and search? calls.